lib/statsd.rb

```diff
@@ -14,14 +14,77 @@
 #   statsd = Statsd.new('localhost').tap{|sd| sd.namespace = 'account'}
 #   statsd.increment 'activate'
 class Statsd
-  class RubyUdpClient
-    attr_reader :key, :sock
+  class BufferedUDPClient < UDPClient
+    @max_packet_size = 511
+    (class << self; attr_accessor :max_packet_size; end)
 
-    def initialize(address, port, key = nil)
+    def initialize(address, port)
+      super(address, port)
+      @buffer = String.new
+    end
+
+    def flush!
+      return unless @buffer.bytesize > 0
+      sock.write(@buffer)
+      @buffer.clear
+    rescue => boom
+      nil
+    end
+
+    def send(msg)
+      flush! if @buffer.bytesize + msg.bytesize > BufferedUDPClient.max_packet_size
+      @buffer << msg
+      @buffer << "\n".freeze
+      nil
```

Comment:
Sending a too-large packet seems like an OK behavior. We could consider sending it immediately rather than waiting for the next `send` call, since we know we're guaranteed to `flush!` on the next `send` anyway.